vm/x86/x86emu/src/emulator/fast_path.rs

```diff
@@ -28,17 +30,17 @@ const PAGE_SIZE: u32 = 4096;
 ///
 /// If the fast path is impossible, returns `None`. The caller should use the
 /// full emulator.
-pub fn emulate_fast_path_set_bit(instruction_bytes: &[u8], state: &mut CpuState) -> Option<u32> {
-    if state.rflags.trap() {
+pub fn emulate_fast_path_set_bit<T: Cpu>(instruction_bytes: &[u8], cpu: &mut T) -> Option<u32> {
```

Comment:
I worry that making this generic is going to cause a significant code size increase...

Reply:
That is a valid concern, but the tradeoff feels forced here, unless I'm missing a trick. The fast_path emulation code resides in x86emu, and without a CpuState object, it requires the cpu functions in order to access registers. To remove the generic parameter, I can only think to pass in all possible relevant registers, at which point this is less of a "fast path".

Is there a quick way for me to measure the code-size increase? Then we can have a concrete idea of what we're up against.

Reply:
Yeah I agree this is required now.